### PR TITLE
Top 5 packages based on GitHub Stars

### LINKED ISSUE #186: Blog Post:  Top 5 Repos based on GitHub Stars and honorable mention?
### Blog Post

Top 5 Repos based on GitHub Stars and honorable mention? 

Short description of each 

Maybe a brief mention to an up and coming package.

### PR DESCRIPTION
**Pull request by gigikenneth**
Thank you for your Pull Request! We have developed this task checklist to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your blog post.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #186 into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update), and make sure the corresponding issue is linked in the Development section on the right hand side
- [x] Run the script from `CICD.R` line by line to first check the spelling in your post and then to make sure your code is compatible with our code-style. Address any incongruences by following the instructions in the file! 
- [x] Choose (possibly several) `tag(s)` or categories from the current list: `c("Metadata", "SDTM", "ADaM", "TLG", "Shiny", "Community", "Conferences", "Submissions", "Technical")` for your blog post. If you cannot find anything that fits your blog post, add your own tag! We occasionally tidy up all `tags` for consistency.
- [x] Add a short description for your blog post in the `description` field at the top of the markdown document.
- [x] Blog post is *short*, *personalized*, *reproducible* and *readable*
- [x] Add a disclaimer at the top of your post, if appropriate (example: **Disclaimer** 
This blog contains opinions that are of the authors alone and do not necessarily reflect the strategy of their respective organizations.)
- [x] Address all merge conflicts and resolve appropriately
- [x] Assign two of us (@bms63, @manciniedoardo, @StefanThoma, @kaz462) as reviewers in the PR.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
whoa...this is way cool!!

**Comment by bms63** [comment]
Could you add the packages needed for this blog to this list in the readme https://github.com/pharmaverse/blog?tab=readme-ov-file#tips-and-tricks and in the gh action that publishes the site please

**Comment by bms63** [comment]
Were you able to build this site locally.  mine is hanging 
image

**Comment by gigikenneth** [comment]
> Were you able to build this site locally. mine is hanging image

Hi! I mistakenly added r tags to the code blocks and that caused the issue. I've taken them out. Hopefully it should work now.

**Comment by gigikenneth** [comment]
The spellcheck errors are mostly coming from the links and some abbreviations.

File: posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+
+Let’s take a look at some features of the package: 
+
+-   [{admiral}](https://github.com/pharmaverse/admiral/) offers **modular functions** for adding variables and records to datasets in a step-by-step manner. This makes it easy to adjust code by adding, removing, or modifying derivations.
+
+-   It provides **template R scripts as a starting point for creating different ADaM datasets.** These templates can be generated using the use_ad_template() function.
+
+-   The package offers a **flexible and efficient way to generate ADaM datasets** from source data, ensuring compliance with CDISC standards and traceability of the data transformation process.
+
+-   **By providing a consistent set of functions and utilities** for ADaM dataset creation, admiral helps to standardize the reporting process across different clinical trials and pharmaceutical companies.
```
Comment (manciniedoardo) [inline posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd:124]: "admiral" is sometimes within curly brackets and sometimes not, and sometimes it's using a link and sometimes not - i would always use curly brackets and a link.

File: posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+-   **Modular design allows for extensions** that focus on specific areas or needs, 3 public packages are currently extending {admiral}, such as [{admiralonco}](https://github.com/pharmaverse/admiralonco), [{admiraloptha}](https://github.com/pharmaverse/admiralophtha) and [{admiralvaccine}](https://github.com/pharmaverse/admiralvaccine). All of them were created from the [admiral template](https://github.com/pharmaverse/admiraltemplate).
```
Comment (manciniedoardo) [inline posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd:126]:
```suggestion
-   **Modular design allows for extensions** that focus on specific areas or needs. Three open-source packages are currently extending {admiral}: [{admiralonco}](https://github.com/pharmaverse/admiralonco), [{admiraloptha}](https://github.com/pharmaverse/admiralophtha) and [{admiralvaccine}](https://github.com/pharmaverse/admiralvaccine). All of them were created from the [admiral template](https://github.com/pharmaverse/admiraltemplate). Two more are in the works: [{admiralpeds}](https://github.com/pharmaverse/admiralpeds) and [{admiralmetabolic}](https://github.com/pharmaverse/admiralmetabolic).
```

File: posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+
+# Display the ADSL dataset
+print(adsl)
+```
+
+**Results**
+
+![](images/admiral-example.png)
+
+**More on Admiral:**
```
Comment (manciniedoardo) [inline posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd:191]: Correct casing on "Admiral"

File: posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+-   [Get Started • admiral](https://pharmaverse.github.io/admiral/articles/admiral.html#setup) 
+
+-   [Recap: Exploring Clinical Submissions with admiral: An R-Based ADaM Solution with Ben Straub](https://www.appsilon.com/post/exploring-clinical-submissions-with-admiral) \
+
+## **{teal}: Interactive Data Exploration**
+
+![](images/teal.png){width="245"}
+
+[**{teal}**](https://github.com/insightsengineering/teal)is an open-source R Shiny framework developed by [Roche](https://www.roche.com/) that enables the creation of interactive data exploration applications for the pharmaceutical industry. [{teal}](https://insightsengineering.github.io/teal/latest-tag/) is particularly well-suited for exploring and analyzing data from clinical trials, enabling researchers and clinicians to quickly identify trends, patterns, and insights.\
```
Comment (manciniedoardo) [inline posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd:207]:
```suggestion
[**{teal}**](https://github.com/insightsengineering/teal) is an open-source R Shiny framework developed by [Roche](https://www.roche.com/) that enables the creation of interactive data exploration applications for the pharmaceutical industry. [{teal}](https://insightsengineering.github.io/teal/latest-tag/) is particularly well-suited for exploring and analyzing data from clinical trials, enabling researchers and clinicians to quickly identify trends, patterns, and insights.\
```

File: posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+source("https://raw.github.com/insightsengineering/teal.gallery/main/_internal/utils/sourceme.R")
+
+# Run the app
+restore_and_run("patient-profile", package_repo = "https://insightsengineering.r-universe.dev")
+
+```
+
+You can also find the [deployed version of the application](https://genentech.shinyapps.io/nest_patient-profile_stable/). 
+
+**Some Resources on {teal}**
```
Comment (manciniedoardo) [inline posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd:241]: Can you also link the teal blog post you wrote?

File: posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd [hunk 6]
```diff
@@ -0,0 +1,359 @@
...
+
+-   [The {riskmetric} Package](https://www.pharmar.org/riskmetric/) 
+
+-   [GitHub - pharmaR/riskmetric: Metrics to evaluate the risk of R packages](https://github.com/pharmaR/riskmetric) 
+
+## **TidyCDISC: Tidying CDISC Data**
+
+![](images/tidyCDISC.png)
+
+[**{tidyCDISC}**](https://github.com/Biogen-Inc/tidyCDISC/)is an open-source R package developed by [Biogen](https://www.biogen.com/) that provides a set of functions for tidying and manipulating [CDISC](https://www.cdisc.org/) (Clinical Data Interchange Standards Consortium) datasets. It aims to simplify the process of working with [CDISC](https://www.cdisc.org/) data by providing an intuitive interface for data transformation tasks and ensuring consistency with the principles of tidy data.
```
Comment (manciniedoardo) [inline posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd:297]:
```suggestion
[**{tidyCDISC}**](https://github.com/Biogen-Inc/tidyCDISC/) is an open-source R package developed by [Biogen](https://www.biogen.com/) that provides a set of functions for tidying and manipulating [CDISC](https://www.cdisc.org/) (Clinical Data Interchange Standards Consortium) datasets. It aims to simplify the process of working with [CDISC](https://www.cdisc.org/) data by providing an intuitive interface for data transformation tasks and ensuring consistency with the principles of tidy data.
```

File: posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+
+### **Features of TidyCDISC**
+
+1.  [{tidyCDISC}](https://github.com/Biogen-Inc/tidyCDISC/) **adheres to the principles of tidy data**, ensuring that CDISC datasets are structured in a consistent and easy-to-work-with format.
+
+2.  The package **supports various CDISC datasets**, including SDTM (Study Data Tabulation Model), ADaM (Analysis Data Model), and SEND (Standard for Exchange of Nonclinical Data).
+
+3.  It provides a set o**f functions for common data transformation tasks,** such as converting between CDISC datasets, handling missing values, and applying CDISC-specific terminology.
```
Comment (manciniedoardo) [inline posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd:305]:
```suggestion
3.  It provides a set of **functions for common data transformation tasks,** such as converting between CDISC datasets, handling missing values, and applying CDISC-specific terminology.
```

File: posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd [hunk 8]
```diff
@@ -0,0 +1,359 @@
...
+
+**More on xportr**
+
+-   [xportr 0.4.0](https://pharmaverse.github.io/blog/posts/2024-03-29_xportr_0_4_0/xportr_0_4_0.html)
+
+-   [Utilities to Output CDISC SDTM/ADaM XPT Files • xportr](https://atorus-research.github.io/xportr/) 
+
+## **Conclusion**
+
+The [pharmaverse](https://pharmaverse.org/) offers a rich ecosystem of tools designed to streamline clinical research workflows, ensuring high-quality data management and reporting. By leveraging packages like [{rtables}](https://github.com/insightsengineering/rtables), [{admiral}](https://github.com/pharmaverse/admiral/tree/main?tab=readme-ov-file), [{teal}](https://github.com/insightsengineering/teal), [{riskmetric}](https://github.com/pharmaR/riskmetric), [{tidyCDISC}](https://github.com/Biogen-Inc/tidyCDISC/), and [{xportr}](https://github.com/atorus-research/xportr/), pharmaceutical companies can enhance their data analysis capabilities, ensure regulatory compliance, and drive innovation in clinical research. **Remember to give the packages a star on GitHub. ⭐**
```
Comment (manciniedoardo) [inline posts/2024-08-01_top_five_pharmaverse_packages/top-five-packages.qmd:357]:
```suggestion
The [pharmaverse](https://pharmaverse.org/) offers a rich ecosystem of tools designed to streamline clinical research workflows, ensuring high-quality data management and reporting. By leveraging packages like [{rtables}](https://github.com/insightsengineering/rtables), [{admiral}](https://github.com/pharmaverse/admiral/tree/main?tab=readme-ov-file), [{teal}](https://github.com/insightsengineering/teal), [{riskmetric}](https://github.com/pharmaR/riskmetric), [{tidyCDISC}](https://github.com/Biogen-Inc/tidyCDISC/), and [{xportr}](https://github.com/atorus-research/xportr/), pharmaceutical companies can enhance their data analysis capabilities, ensure regulatory compliance, and drive innovation in clinical research. **Remember to give the packages that you use and value a star on GitHub. ⭐**
```

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Great post @gigikenneth ! I left some comments. Spelling can be fixed by adding words to `inst/wordlist`.

**Comment by gigikenneth** [comment]
Hi @manciniedoardo , please I need help. the spellcheck is failing for one word. I've updated it in the doc but it's not working 🫠

**Comment by manciniedoardo** [comment]
> Hi @manciniedoardo , please I need help. the spellcheck is failing for one word. I've updated it in the doc but it's not working 🫠

There was a space after "XPORT". I removed that and took the opportunity to re-sort the WORDLIST as it was getting out of hand.

**Comment by manciniedoardo** [comment]
@gigikenneth @bms63 could we please make an effort to merge in this week in order to fit this post into the Aug newsletter? thanks 😄

File: posts/2024-12-01_my__test__blog/my__test__blog.qmd [hunk 9]
```diff
@@ -0,0 +1,37 @@
+---
+title: "My Test Blog"
```
Comment (manciniedoardo) [inline posts/2024-12-01_my__test__blog/my__test__blog.qmd:2]: Is this file needed?

**Review by manciniedoardo — COMMENTED** [review]
I think there is a spurious blog post here? My test blog?

File: posts/2024-12-01_my__test__blog/my__test__blog.qmd (hunk 9 above)
Comment (gigikenneth) [reply posts/2024-12-01_my__test__blog/my__test__blog.qmd:2]: omg noooo....I don't know how it got in there. It was a test blog post for a draft. 😭

**Comment by gigikenneth** [comment]
> I think there is a spurious blog post here? My test blog?

@manciniedoardo I've deleted the test blog 🙏🏾

**Review by manciniedoardo — APPROVED** [review]
